totes

```diff
@@ -21,30 +30,42 @@ get_note_to_render() {
       echo "The notes directory - $notes_path - does not exist."
       # shellcheck disable=2016
       echo 'Try running the `install.sh` script again'
-      # checks if this script is in a repo (should be true if it is symlinked)
+      # checks if thlis script is in a repo (should be true if it is symlinked)
       if git rev-parse HEAD > /dev/null 2>&1; then
         # display full path to install script if we can find it
         echo "path: $(git rev-parse --show-toplevel)/install.sh"
       fi
       exit 1
     fi
 
+    suppressed_files_registry_spec="$(get_suppressed_files_registry_spec)"
+    if ! [[ -f "$suppressed_files_registry_spec" ]]; then
+      # the suppressed_file_registry must not be empty
+      echo "dummy-text-required-here" > "$suppressed_files_registry_spec"
+    fi
+
     # ref - https://github.com/koalaman/shellcheck/issues/1247
     # shellcheck disable=2012
-    note_to_render="$(ls -At -- "$notes_path"/*.md | tail -n 1)"
+    # shellcheck disable=2010
+    note_to_render="$(ls -At -- "$notes_path"/*.md | grep -vFf "$suppressed_files_registry_spec" | tail -n 1)"
```

Comment:
fwiw, I'm working on issue #8, where I'm planning to change this line to the following:

`      ls -Art -- "$notes_path"/**/*.md | grep -m 1 -vFf "$suppressed_files_registry_spec"`

That is, reverse the ls listing, glob into subdirectories, and only return the first grep match, which obviates the need to pipe through 'tail' (while potentially being faster)

Reply:
what do you think of refactoring this onto separate lines and changing how we guard the `grep`? something like:

```bash
all_notes="$(ls -At -- "$notes_path"/*.md)"
if [[ -f "$suppressed_files_registry_spec" ]]; then
  # TODO: add context on these flags and/or where this solution came from
  active_notes="$(grep -vFf "$suppressed_files_registry_spec" <<< "$all_notes")"
fi
note_to_render="$(tail -n 1 <<< "${active_notes:-$all_notes}")"
```

Reply:
I do like that better.  Knowing there are other PR's that touch these lines "in the queue", I'd rather wait for the dust to settle slightly, and make this suggested change in a later PR

Reply:
In general I think needing to rebase existing PRs on new changes from other merges should be an expected, not uncommon practice. For example I merged a few PRs today that are creating conflicts and will require a rebase of this PR on `master`. If you'd rather do it in a follow-up though that's totally fine by me.